tests/ui/types_fn_to_int.rs

```diff
@@ -0,0 +1,15 @@
+enum Foo {
+    A(usize),
+    B
+}
+
+fn bar() -> i32 {
+    0i32
+}
+
+fn main() {
+    let x = Foo::A;
+    let y = x as i32;
```

Comment:
Can you also add a check for `Foo::A as i32` directly without an intermediate variable?